### DISCUSSION TITLE
Github forgets about my custom domain every time I update the website

### DISCUSSION THREAD

**Discussion by lucypero** [community/community#21549] [Other Feature Feedback, Questions, & Ideas]
<p>It resets to the default <a href="http://xxx.github.io" rel="nofollow noopener">xxx.github.io</a> URL every time I update the site. I have to input the custom domain on the repository settings every single time I update the website. Any advice?</p>

**Comment by RyanZim** [comment, marked as answer]
<p>When you configure the custom domain in the settings, it adds a CNAME file to the pages root. You must be force-pushing when you update your site. If you add a CNAME file containing the custom domain name to your pages root (or pull in the commit GH makes when you set the custom domain in Settings), you shouldn’t have this problem.</p>

**Reply by negative0** [reply]
If you are using [gh-pages](https://www.npmjs.com/package/gh-pages) to deploy your site, you can modify your package.json deploy [command](https://www.npmjs.com/package/gh-pages#deploying-to-github-pages-with-custom-domain) to:

```
echo your_cutom_domain.online > ./build/CNAME && gh-pages -d build"
```

**Comment by Flarsystem** [comment]
<blockquote>
<p><a class="mention" href="/ryanzim">@ryanzim</a> wrote:</p>
</blockquote>
<blockquote>
<p>When you configure the custom domain in the settings, it adds a CNAME file to the pages root. You must be force-pushing when you update your site. If you add a CNAME file containing the custom domain name to your pages root (or pull in the commit GH makes when you set the custom domain in Settings), you shouldn’t have this problem.</p>
</blockquote>

**Comment by nadiajoyce** [comment]
<p><a class="mention" href="/flarsystem">@flarsystem</a> It looks like you started a comment here with a quote, but then the rest of your content didn’t come through. Was there more you meant to add here?</p>

**Comment by ssbarnea** [comment]
That is insane as any force push, even on a feature branch will break the entire website. Even if the default branch is not touched and the correct `CNAME` file remains there, the site is automatically unpublished from github pages.

One can easily conclude that gh-pages cannot really be considered reliable as it can suddenly vanish.

**Reply by alpritt** [reply]
I've just come across this. For clarity, commit a copy of the CNAME file to your build directory (in my case /public/CNAME). By the looks of it, each push is going to cause a rebuild of the contents of the gh-pages branch based on your pubic build directory. So if the CNAME file isn't committed to that public directory on your main branch, it will be wiped out.

**Comment by Matary1** [comment]
Hello I have question

**Reply by akuzni2** [reply]
https://nohello.net
